r-package/grf/R/survival_forest.R

```diff
@@ -0,0 +1,206 @@
+#'  K elements of the cluster to the tree-growing procedure. When estimating average treatment effects,
+#'  each observation is given weight 1/cluster size, so that the total weight of each cluster is the
+#'  same. Note that, if this argument is FALSE, sample weights may also be directly adjusted via the
+#'  sample.weights argument. If this argument is TRUE, sample.weights must be set to NULL. Default is
+#'  FALSE.
+#' @param sample.fraction Fraction of the data used to build each tree.
+#'                        Note: If honesty = TRUE, these subsamples will
+#'                        further be cut by a factor of honesty.fraction. Default is 0.5.
+#' @param mtry Number of variables tried for each split. Default is
+#'             \eqn{\sqrt p + 20} where p is the number of variables.
+#' @param min.node.size A target for the minimum number of observations in each tree leaf. Note that nodes
+#'                      with size smaller than min.node.size can occur, as in the original randomForest package.
+#'                      Default is 5.
+#' @param honesty Whether to use honest splitting (i.e., sub-sample splitting). Default is TRUE.
+#'  For a detailed description of honesty, honesty.fraction, honesty.prune.leaves, and recommendations for
+#'  parameter tuning, see the grf
+#'  \href{https://grf-labs.github.io/grf/REFERENCE.html#honesty-honesty-fraction-honesty-prune-leaves}{algorithm reference}.
+#' @param honesty.fraction The fraction of data that will be used for determining splits if honesty = TRUE. Corresponds
+#'                         to set J1 in the notation of the paper. Default is 0.5 (i.e. half of the data is used for
+#'                         determining splits).
+#' @param honesty.prune.leaves If TRUE, prunes the estimation sample tree such that no leaves
+#'  are empty. If FALSE, keep the same tree as determined in the splits sample (if an empty leave is encountered, that
+#'  tree is skipped and does not contribute to the estimate). Setting this to FALSE may improve performance on
+#'  small/marginally powered data, but requires more trees (note: tuning does not adjust the number of trees).
+#'  Only applies if honesty is enabled. Default is TRUE.
+#' @param alpha A tuning parameter that controls the maximum imbalance of a split. Default is 0.05
+#'  (meaning the count of failures on each side of a split has to be at least 5 % of the total observation count in a node)
+#' @param compute.oob.predictions Whether OOB predictions on training set should be precomputed. Default is TRUE.
+#' @param num.threads Number of threads used in training. By default, the number of threads is set
+#'                    to the maximum hardware concurrency.
+#' @param seed The seed of the C++ random number generator.
+#'
+#' @return A trained survival_forest forest object. The attribute `failure.times` contains the unique failure
+#'  times in the data set.
+#'
+#' @examples
+#' \dontrun{
+#' # Train a standard survival forest.
```

Comment:
It would be nice to give the user an example where the user can actually see what the forest did. For example, one thing we could do is along these lines:

```R
n <- 2000
p <- 5
X <- matrix(rnorm(n * p), n, p)
latent.time <- exp(0.5 * X[, 1]) * rexp(n)
censor.time <- 2 * rexp(n)
Y <- pmin(latent.time, censor.time)
D <- as.integer(latent.time <= censor.time)
s.forest <- survival_forest(X, Y, D)

# Predict using the forest.
X.test <- matrix(0, 3, p)
X.test[, 1] <- seq(-2, 2, length.out = 3)
s.pred <- predict(s.forest, X.test)

cols = RColorBrewer::brewer.pal(3, "Set1")
plot(NA, NA, xlab = "failure time", ylab = "survival function",
     xlim = range(s.pred$failure.times),
     ylim = c(0, max(s.pred$predictions)))

for(iter in 1:3) {
  lines(s.pred$failure.times, s.pred$predictions[iter,], col = cols[iter], lwd = 2)
  s.true = exp(-s.pred$failure.times / exp(0.5 * X.test[iter,1]))
  lines(s.pred$failure.times, s.true, col = cols[iter], lty = 2, lwd = 2)
}
```